patches/minecraft/net/minecraft/client/renderer/texture/LayeredColorMaskTexture.java.patch

```diff
@@ -0,0 +1,18 @@
+--- ../src-base/minecraft/net/minecraft/client/renderer/texture/LayeredColorMaskTexture.java
++++ ../src-work/minecraft/net/minecraft/client/renderer/texture/LayeredColorMaskTexture.java
+@@ -71,9 +71,14 @@
+ 
+                         if (s != null)
+                         {
++                            BufferedImage bufferedimageTemp = net.minecraftforge.client.MinecraftForgeClient.getBannerImage(new ResourceLocation(s));
++                            if (bufferedimageTemp == null)
++                            {
+                             iresource1 = p_110551_1_.func_110536_a(new ResourceLocation(s));
+                             BufferedImage bufferedimage2 = TextureUtil.func_177053_a(iresource1.func_110527_b());
```

Comment:
I think this patch is a bit confusing, so it might be a bit cleaner if you stuff the two lines of vanilla logic into `MinecraftForgeClient.getBannerImage` and basically replace this section with one line doing 
`BufferedImage bufferedimage2 = <getBannerImage>`

`iresource1` will remain null and things should work fine as long as you have similar error handling/closing logic in your method.

Reply:
Can do. I was trying to leave the original code untouched, but it does make it pretty messy. I'll move it. :)

Reply:
Note that doing this means the function cannot return null, so the reverse of the other change will be applied. Also, it must be marked as able to throw an IOException, as the vanilla lines might do so.